src/proto/meta_protocol.rs

```diff
@@ -84,28 +98,16 @@ impl MetaProtocol for Client {
 
         self.conn.flush().await?;
 
-        if quiet_mode {
-            match self.drive_receive(parse_meta_get_response).await? {
-                MetaResponse::Status(Status::NoOp) => Ok(None),
-                MetaResponse::Status(s) => Err(s.into()),
-                MetaResponse::Data(d) => d
-                    .map(|mut items| {
-                        let item = items.remove(0);
-                        Ok(item)
-                    })
-                    .transpose(),
-            }
-        } else {
-            match self.drive_receive(parse_meta_get_response).await? {
-                MetaResponse::Status(Status::NotFound) => Ok(None),
-                MetaResponse::Status(s) => Err(s.into()),
-                MetaResponse::Data(d) => d
-                    .map(|mut items| {
-                        let item = items.remove(0);
-                        Ok(item)
-                    })
-                    .transpose(),
```

Comment:
This conditional wasn't actually required in this case, the match arms can be condensed together for the same function.